### PR TITLE
DH Key Exchange

### PR DESCRIPTION
**Pull request by dspearson**
This is some early work to expose the new Key Exchange functionality in libsodium.

Currently, using call! does not work. Included is a snippet that may aid in debugging. Manually calling the method on sodium appears functional.

Extremely bare-bones currently; no tests, not fully implemented, and not exactly functional.

### CODE REVIEW

**Comment by dspearson** [comment]
There's still some work to do; the kx fns don't do in-place mod, so I need to add -to-buf!, and the bug related to the use of call! still exists.

**Comment by dspearson** [comment]
I think I've done all I can with this for now. I'll await commentary.

File: dev-resources/vectors/kx/alice-public-key [hunk 1]
```diff
@@ -0,0 +1 @@
+8520f0098930a754748b7ddcb43ef75a0dbf3a0d26381af4eba4a98eaa9b4e6a
```
Comment (lvh) [inline dev-resources/vectors/kx/alice-public-key]: Did these test vectors come from kalium? If so, please add a README file to the vector folder to about where you got these from. (Ideally it'd be a shell script to download the vectors from a known-good source, but I appreciate that is not always possible :-))

Comment (dspearson) [reply dev-resources/vectors/kx/alice-public-key]: I added some C code to show how I generate the vectors, and a note about where they come from. Unfortunately it also showed that my tests were not exhaustive, or are incorrect. I'll try and rework it when I have time.

**Comment by dspearson** [comment]
I believe this is ready for review.

**Pull request merged**